ecs-init/config/common.go

```diff
@@ -99,3 +111,42 @@ func DockerUnixSocket() (string, bool) {
 	// eg: Docker daemon restart
 	return "/var/run", false
 }
+
+// Find AZ information from Metadata. If error return a blank result
+func EC2MetadataAZ () string {
+	sessionInstance := session.Must(session.NewSession())
+	metadata := ec2metadata.New(sessionInstance)
+	azName, err := metadata.GetMetadata("placement/availability-zone")
+
+	if err != nil {
+		return ""
+	}
+
+	return azName
+}
+
+//convert AZ name to Region name or default if blank is returned
+func AZToRegionName (azName string) string {
+	if azName == "" {
+		return DefaultRegionName
+	}
+
+	return azName[0:len(azName)-1]
+}
+
+// Get Bucket from list of S3 Buckets by region name or default if key is not found
+func GetS3BucketMapByRegion(regionName string) string {
+	val, exists := S3BucketMap[regionName]
+	if !exists {
+		return S3BucketMap[DefaultRegionName]
+	}
+
+	return val
+}
+
+// Retreive tarball URL from S3 Bucket list by searching for region name
+func FindTarballUrl () string {
```

Comment:
Why can't this be in `AgentRemoteTarball` so that you can get rid of this method? Right now `AgentRemoteTarball` is calling this directly and doing nothing else. Let's avoid that pattern.